tunnel/src/main/java/com/wireguard/android/backend/GoBackend.java

```diff
@@ -219,7 +219,11 @@ private void setStateInternal(final Tunnel tunnel, @Nullable final Config config
             final VpnService service;
             if (!vpnService.isDone()) {
                 Log.d(TAG, "Requesting to start VpnService");
-                context.startService(new Intent(context, VpnService.class));
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    context.startForegroundService(new Intent(context, VpnService.class));
```

Comment:
If you are starting the service  as a foreground service, you should accompany it with a notification. 
Also, according to the guidelines, it is okay to start the VPN service as a background service, but you must promote it to foreground by showing the notification using the Service#setForeground method.

https://developer.android.com/guide/topics/connectivity/vpn#starting_a_service

Thoughts?